x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/rules_stats_collector.ts

```diff
@@ -0,0 +1,261 @@
+            },
+          },
+        },
+        rule_version: {
+          top_metrics: {
+            metrics: {
+              field: 'rule.version',
+            },
+            size: 1,
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+        posture_type: {
+          top_metrics: {
+            metrics: {
+              field: 'rule.benchmark.posture_type',
+            },
+            size: 1,
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+        rule_number: {
+          top_metrics: {
+            metrics: {
+              field: 'rule.benchmark.rule_number',
+            },
+            size: 1,
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+        benchmark_id: {
+          top_metrics: {
+            metrics: {
+              field: 'rule.benchmark.id',
+            },
+            size: 1,
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+        benchmark_version: {
+          top_metrics: {
+            metrics: {
+              field: 'rule.benchmark.version',
+            },
+            size: 1,
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+        benchmark_name: {
+          top_metrics: {
+            metrics: {
+              field: 'rule.benchmark.name',
+            },
+            size: 1,
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+        passed_findings_count: {
+          filter: {
+            bool: {
+              filter: [
+                {
+                  bool: {
+                    should: [
+                      {
+                        term: {
+                          'result.evaluation': 'passed',
+                        },
+                      },
+                    ],
+                    minimum_should_match: 1,
+                  },
+                },
+              ],
+            },
+          },
+        },
+        failed_findings_count: {
+          filter: {
+            bool: {
+              filter: [
+                {
+                  bool: {
+                    should: [
+                      {
+                        term: {
+                          'result.evaluation': 'failed',
+                        },
+                      },
+                    ],
+                    minimum_should_match: 1,
+                  },
+                },
+              ],
+            },
+          },
+        },
+      },
+    },
+  },
+
+  size: 0,
+  _source: false,
+});
+
+const getCspmRulesStats = (aggregatedRulesStats: RulesStats, logger: Logger): CspmRulesStats[] => {
+  const rules = aggregatedRulesStats.rules.buckets;
+
+  const cspmRulesStats = rules.map((rule) => ({
+    rule_id: rule.key,
+    rule_name: rule.rule_name.top[0].metrics['rule.name'],
```

Comment:
not sure how likely this is but if `top` is empty, this line and others would throw